client/blocks/upsell-nudge/index.jsx

```diff
@@ -90,9 +91,16 @@ export const UpsellNudge = ( {
 	}
 
 	if ( ! href && siteSlug && canUserUpgrade ) {
-		href = addQueryArgs( { feature, plan }, `/plans/${ siteSlug }` );
-		if ( customerType ) {
-			href = `/plans/${ siteSlug }?customerType=${ customerType }`;
+		const currentPlanSlug = currentPlan?.productSlug;
+
+		// Redirect to the checkout page if the user is on a Premium or Personal plan since the can't view the overhauled Plans page.
+		if ( isPremiumPlan( currentPlanSlug ) || isPersonalPlan( currentPlanSlug ) ) {
```

Comment:
Maybe `isBloggerPlan()` also?  We do still have users on that, and I suppose it would probably experience the same bug.